### DISCUSSION TITLE
Custom OTA for Xiaomi LYWSD03MMC device converted to Zigbee

### DISCUSSION THREAD

**Discussion by bipsendk** [Koenkk/zigbee2mqtt#23082] [Devices]
Hi

 Is there a way to "force" a custom OTA update to an Xiaomi LYWSD03MMC device, which currently shows up as "Custom Devices (DiY)" to use another variant of the firmware, so that the device shows up as Xiaomi LYWSD03MMC-z (I think both Bluetooth and Zigbee are enabled on this file/version) ?
If yes - how can I do it ? It will probably be a lot easier than having to disassemble the device and update/re-flash via serial port...

Regards
Brian

**Comment by Virgun** [comment]
Same, trying to force an update but getting manufacturer ID mismatch (well, as it is supposed to be) but can't find any way to bypass it.

There is an easier way if you want to:
Inside pvvx library there is "devbis" folder.

Open readme.txt and find firmware which you need in order to update to pvvx firmware.

Add to "configuration.yaml"
```
ota:
    zigbee_ota_override_index_location: /path/to/file.json
```
Download, place somewhere inside z2m and override OTA index file https://www.zigbee2mqtt.io/guide/usage/ota_updates.html#local-ota-index-and-firmware-files

All you need to place inside override OTA index is:
```
[
    {
        "url": "./path/to/my/image.ota"
    }
]
```

**Comment by mrmaximas** [comment]
@Virgun this will help you https://github.com/pvvx/ZigbeeTLc/tree/master/bin/devbis

**Comment by CrazYoshi** [comment]
How to get the device in download mode?
I'm trying to force the firmware upgrade using the custom firmware option in the Z2M OTA but I'm getting this error:
`3/14/2026, 9:44:13 PM] z2m: OTA update of 'sensori_dhtstudio' failed (Device 0xa4c1387f6ca88d92 did not start/finish firmware download after being notified. (Timeout - 32123 - 1 - undefined - 25 - 3 after 150000ms))  `
